libs/ledger/src/chain/main_chain.cpp

```diff
@@ -263,6 +265,11 @@ bool MainChain::RemoveBlock(BlockHash hash)
 {
   FETCH_LOCK(lock_);
 
+  // Step 0. Manually set heaviest to a block we still know is valid
+  auto block_before_one_to_del = GetBlock(hash);
+  heaviest_.weight             = block_before_one_to_del->body.weight;
+  heaviest_.hash               = block_before_one_to_del->body.hash;
```

Comment:
I think this needs
```
heaviest_.Update(*block_before_one_to_del);
```

Reply:
I believe this will only update if the proposed block is heavier - since it is not in this case I have done it manually. Perhaps I should have reset the heaviest tip then called update would be cleaner